### PR TITLE
Added shallow depth clone and gc pruning

### PR DESCRIPTION
**Pull request by mattpaletta**
#7046 
I added shallow clone in Git, and git pruning to remove local history where I felt appropriate.  I may have missed some, feel free to comment.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -62,6 +62,8 @@ begin
       puts Dir.pwd
       sh "git checkout #{specs_branch}"
       sh 'git pull'
+      sh 'git pull'
```
Comment (dnkoutso) [inline Rakefile]: this is duplicated with the line above

**Comment by dnkoutso** [comment]
Sorry this has been discussed before. It is intentional that we do not do shallow clones anymore.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Added shallow depth clone and gc pruning  
  [mattpaletta](https://github.com/mattpaletta)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by mattpaletta** [comment]
OK, I just read this PR: https://github.com/CocoaPods/CocoaPods/issues/5016
I assume this is the kind of discussion you are referring to.  Could I remove the shallow clone, but keep the `git gc`, so the repo takes up less local space on the machine, without the stress on Github servers?

**Comment by mattpaletta** [comment]
OK, I tried to address your comments and removed the shallow clone, but kept the git GC, and got the travis tests working.  Feel free to merge it if you think it would benefit the project in the short-term, if not, that's fine too :)

File: Rakefile [hunk 2]
```diff
@@ -62,6 +62,7 @@ begin
       puts Dir.pwd
       sh "git checkout #{specs_branch}"
       sh 'git pull'
+      sh 'git gc --prune=now'
```
Comment (amorde) [inline Rakefile]: Running this on my local copy took ~20 minutes and kept a CPU pegged at 100% usage, but only saved ~50MB. I don't think it's worth the setup time

File: lib/cocoapods/sources_manager.rb [hunk 3]
```diff
@@ -114,6 +114,7 @@ def update_git_repo(show_output = false)
         git!(args)
         current_branch = git!(%W(-C #{repo} rev-parse --abbrev-ref HEAD)).strip
         git!(%W(-C #{repo} reset --hard origin/#{current_branch}))
+        system 'git gc --auto'
```
Comment (amorde) [inline lib/cocoapods/sources_manager.rb]: `git gc --auto` only garbage collects if git [determines it is required](https://git-scm.com/docs/git-gc) - I tried running this before running `git gc --prune` and it exited without taking any action. Will this actually have any effect?

File: lib/cocoapods/sources_manager.rb [hunk 4]
```diff
@@ -128,6 +129,7 @@ def update_git_repo(show_output = false)
       if repo.join('.git', 'shallow').file?
         UI.info "Performing a deep fetch of the `#{name}` specs repo to improve future performance" do
           git!(%W(-C #{repo} fetch --unshallow))
+          system 'git gc --auto'
```
Comment (amorde) [inline lib/cocoapods/sources_manager.rb]: for consistency these should use the `git!` function like above

```ruby
git!(%W(gc --auto))
```

**Comment by orta** [comment]
Yeah, I don't think this is worth it IMO - we should probably close it

**Pull request closed without merging**